### ISSUE TITLE
Sounds and Conveyances[Edit, was: Media/Sound support]

### ISSUE THREAD

**Issue by mjy** [SpeciesFileGroup/taxonworks#244]
* Identical to Image and Document
* Probably `Media::Sound`, `Media::Video`, etc.

**Comment by klausriede** [comment]
following my initial question, I now moved to this thread about media/sound support. I learned that there is still no data model for sound. I think the present available (old) OSF data model is fine, it just needs updating with some additional field. What do you think?

**Comment by mjy** [comment]
@klausriede We will try and follow the same semantics as for Images and Documents.  This means that the basic Sound file will be just the sound file and metadata about that file.  What the sounds is "about" is defined by linking that file to it target.

For images this is what an image `depicts`, for files/documents this is `documentation`.  The name of the relationship between the media file itself, and the object/things it describes needs a name, I haven't come up with one, if you have suggestions that would be great.

If there are attributes that you want to have considered that are not in SFs listing them here would be good.  We might not be able to cover them all, but we have lots of extensibility with our already existing annotation system.

**Comment by klausriede** [comment]
Most information will probably be contained in the metadata, in fact, I would prefer that the soundfile itself is containing metadata, however, there are no standards, as in pics. I am not sure if I understand your terminology, I am used to SQL tables and could provide you with the respective data model we already developed in DORSA and SYSTAX, which you probably can translate into your terminology. In any case I would like to understand your data model and terminology, is there any white paper about this?

**Comment by mjy** [comment]
There is no whitepaper on this, sorry (another thing TODO). Conceptually the sound record should contain no biological metadata.  The general concept is that one soundfile file/record may describe one or more entities/"objects".  This is easiest to understand with images and taxa.  One image may depict both a giraffe and a lion.  This means we use 5 records, the image, the giraffe, the lion, the first link (depiction) between image and giraffe, and the second depiction between image and lion.  The row in the image table does not include fields for species (lion, giraffe), their parts, or any other biological setup.

We need the same setup for sound files.  For example one sound file may contain 5 insects in it.  So we can say that this sound file "has_noises_for" (or some such word) OTUs a, b, c, etc.    Similarly it may have attributes of the noise for an Otu (some type of call [mating, warning]), these attributes are akin to determinations, they are asserted by the user, they are not part of the bits of the music themselves.  In other words, in the databse itself, the sound table has just information about the file itself (how big it is, what format it is, etc.), nothing about biology is stored in that table.  We (the currators) then take this file and link it to biological assertions (this sounds has_noises_for some species).

**Comment by klausriede** [comment]
This is a rather ambitious data model, which certainly suits my ideas of using soundscapes for acoustic  monitoring of Orthoptera (corresponding to acoustic scenes fig 4 in Riede 2018, [https://jor.pensoft.net/article/23700/](https://jor.pensoft.net/article/23700/) ) However, right now we manage individual recordings in OSF, which basically is a 1:n relation between a specimen and n recordings (for example at distinct temperatures, in distinct behavioural situations),  a much simpler model. Several more fields as in present OSF are needed. How do we proceed? Before any programming we certainly do need a data model and user specification, is there any document about this? I tried to find documentation via taxonworks and on https://github.com/SpeciesFileGroup/taxonworks_doc . The scope sounds promising, and I really would like to try annotation of soundfiles, and import of simple acoustic features into data matrices.

**Comment by mjy** [comment]
@klausriede As a side note has anyone actually reached out to the ornithologists to see if their resources could be re-used for insects, for example files and metadata could be stored on one sound-specalists site, and then linked to a project in TaxonWorks where additional data could be tracked.

**Comment by mjy** [comment]
FYI significantly more information on the existing models in TaxonnWorks, from a technical perspective is at the autogenerated documentation site http://rdoc.taxonworks.org/frames.html#!file.README.html.

**Comment by mjy** [comment]
@klausriede Thinking a little more I suspect we have everything in place to link songs to data points already, asside from the linking model I desribed above.  Spend some time looking at the Matrix hub task- you can create full matrices of data of various types.  Every Obersvation in that matrix will be linkable to a sound, the coding and mechanism for this is already done for images, it's a very straightforward extension to implement sounds, its just something that take time.

**Comment by klausriede** [comment]
Starting with the sidenote, I am in touch with https://github.com/BioAcoustica and @edwbaker, Bioacoustica houses ALL soundfiles. Again, some thoughts about connecting sound databases as a distributed data warehouse in my paper. I will look into the Matrix, in fact this was my first idea because I have a large table of OTUs and sound parameters ready for upload. Any possibilities to upload tables as a Matrix?

**Comment by mjy** [comment]
If your data are in Mesquite valid nexus format then we have an importer
library that will ultimately be avilable.  We will likely have other
options to do batch load data down there as well.

M

**Comment by edwbaker** [comment]
@mjy - will you be doing the TaxonWorks workshop At ICO in Agadir? If so perhaps we can discuss briefly then? (I am committed to helping a different workshop though)

**Comment by mjy** [comment]
@edwbaker I won't be there.  IIRC Maria Marta will be doing a presentation.

**Comment by LocoDelAssembly** [comment]
Possible player? https://wavesurfer-js.org/examples/ (looks like it can even perform FFT)

**Comment by klausriede** [comment]
looks interesting! However, I sadly remember Java applications which do not run anymore (you need to install an old Browser version). For the time being it would be sufficient to download files and use your own software, however, bulk (basket) download of several files would be useful. Do we have any thread or rolling document to jot down user requirements?

**Comment by mjy** [comment]
This is javascript, not java, very different.   Nevertheless we should see
how active the package maintenance is.

**Comment by edwbaker** [comment]
This is the player we use for BioAcoustica - see e.g. http://bio.acousti.ca/content/myopsaltaleonakindon

It has been well maintained for as long as we have been using it (approaching 4 years I think)

**Comment by jlpereira** [comment]
I agree with @LocoDelAssembly . Wavesurfer is an awesome library with much features and plugins, version 2 works with webpack. It should be pretty easy to use it in a vue interface

**Comment by mjy** [comment]
I think we'll use the following terms for the otherwise well understand models:

Sounds (as to Image)
Conveyance (as to Depiction)

A Conveyance conveys_some Sound from an <OTU/CollectionObject/FieldOccurrence>

Perhaps also Portrayals as a  top level class for Depictions and Coveyances.

**Comment by klausriede** [comment]
please provide some material to understand this! I repeat my simple question: will multimedia objects be stored inside the database? Or outside, with URNs? I cannot see any major difference between multimedia image, sound or video, so why not handle all of them?

**Comment by mjy** [comment]
Notes for the 244 branch

* [ ] mime-type validation
* [ ] matrix implementation
* [ ] origin implementation (UI)
* [ ] basic player to confirm content

**Comment by klausriede** [comment]
I see this has been closed - does this mean that by now we have a sound data model and we now can add sounds in OSF?

**Comment by jlpereira** [comment]
Yes, it will be available in the next release

**Comment by klausriede** [comment]
Sounds like great news! About when, and is there any information about the underlying datamodel?
